wave/src/User.php

```diff
@@ -56,18 +59,29 @@ public function onTrial(){
         return true;
     }
 
-    public function subscribed($plan){
+    public function subscribed($plan = 'default', $price = null) {
 
-        $plan = Plan::where('slug', $plan)->first();
+        if(env('CASHIER_VENDOR') == 'stripe') {
+            $subscription = $this->subscription($plan);
 
-        // if the user is an admin they automatically have access to the default plan
-        if(isset($plan->default) && $plan->default && $this->hasRole('admin')) return true;
+            if (!$subscription || !$subscription->valid()) {
+                return false;
+            }
 
-        if(isset($plan->slug) && $this->hasRole($plan->slug)){
-            return true;
+            return !$price || $subscription->hasPrice($price);
+        } else {
```

Comment:
Same here, the `else` shouldn't be needed since we `return` early if the `if` statement is true.